presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -195,16 +202,23 @@ private boolean joinCurrentPosition()
     {
         // while we have a position on lookup side to join against...
         while (joinPosition >= 0) {
-            pageBuilder.declarePosition();
-
-            // write probe columns
-            probe.appendTo(pageBuilder);
-
-            // write build columns
-            lookupSource.appendTo(joinPosition, pageBuilder, probe.getOutputChannelCount());
+            lookupPositionsConsidered++;
+            if (lookupSource.isJoinPositionEligible(joinPosition, probe.getPosition(), probe.getPage())) {
```

Comment:
I will run it. I notice that the benchmark has no filter function. Any suggestions on running it with a filter function? I could modify the benchmark to add a simple filter function I guess.